main.rb

```diff
@@ -0,0 +1,95 @@
+module Enumerable
+
+    def my_each
+    unless !block_given?
+      for item in self
+        yield(item)
+      end
+    end
+      return self
+    end
+  
+    def my_each_with_index
+    unless !block_given?
+      for item_index in (0...self.size)
+        yield(self[item_index], item_index)
+      end
+    end
+      return self
+    end
+  
+    def my_select
+    unless !block_given?
+      new_items = []
+      my_each do |item|
+        new_items.push(item) if yield(item)
+      end
+      return new_items
+    end
+      return self
+    end
+  
+    def my_all?
+    return true if !block_given?
+    items_exist = true
+      my_each do |item|
+            items_exist = false if !yield(item)
+      end
+    items_exist
+    end
+  
+    def my_count
+      return self.size if !block_given?
+      count = 0
+      my_select do |item|
+          count+=1 if yield(item)
+      end
+      count
+    end
+  
+    def my_any?
+      result = false
+      my_each do |item|
+          result = true if yield(item)
+      end
+      result
+    end
+  
+    def my_none?
+      result = true
+      unless !block_given?
+        my_each do |item|
+            result = false if yield(item)
+        end
+      end
+      return result
+    end
```

Comment:
`my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- when no block or argument is given returns true only if none of the collection members is true
```ruby
[false, nil, false].my_none? #should return true
[1, 'demo', 2.2].my_none? #should return false
```
- when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2, 3].my_none?(String)  #should return true
['1', '2', '3'].my_none?(String)  #should return false
```
- when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/)   #should return true
%w[dog door rod blade].my_none?(/d/)   #should return false
```
- when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[1, 2, 4].my_none?(3) #should return true
[1, 2, 3].my_none?(3) #should return false
```